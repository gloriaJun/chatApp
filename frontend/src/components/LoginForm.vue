<template>
  <v-form
    ref="form"
    v-model="valid"
  >
    <v-text-field
      v-model="username"
      :rules="usernameRules"
      name="login"
      label="Username"
      placeholder="example"
      required
    ></v-text-field>

    <v-btn
      :disabled="!valid"
      color="primary"
      block
      @click="submit"
    >
      Login
    </v-btn>
  </v-form>
</template>

<script>
export default {
  data: () => ({
    valid: false,
    username: '',
    usernameRules: [
      v => !!v || 'username is required',
    ],
  }),
  methods: {
    submit() {
      if (this.$refs.form.validate()) {
        this.$emit('submit-login', { username: this.username });
      }
    },
  },
};
</script>

<style>

</style>
