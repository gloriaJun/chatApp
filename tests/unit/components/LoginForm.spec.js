import {
  mount,
} from '@vue/test-utils';
import '../test-setup';

import LoginForm from '@/components/LoginForm.vue';

describe('LoginForm.vue', () => {
  it('snapshot check', () => {
    const wrapper = mount(LoginForm);

    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it('when rendering button should be inactive', () => {
    const wrapper = mount(LoginForm);
    const el = wrapper.find('.v-btn');

    expect(el.attributes('disabled')).toEqual('disabled');
  });


  it('when input set value, button should be active', () => {
    const wrapper = mount(LoginForm);
    const username = 'user01';

    const inputWrapper = wrapper.find('.v-input');
    const btn = wrapper.find('.v-btn');

    inputWrapper.find('input').setValue(username);

    expect(btn.attributes('disabled')).toBeUndefined();
  });
});
