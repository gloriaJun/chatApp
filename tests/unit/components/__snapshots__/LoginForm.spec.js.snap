// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm.vue snapshot check 1`] = `
<form
  class="v-form"
  novalidate="novalidate"
>
  <div
    class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--placeholder theme--light"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <div
          class="v-text-field__slot"
        >
          <input
            aria-label="Username"
            name="login"
            placeholder="Input Username"
            required="required"
            type="text"
          />
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light"
        >
          <div
            class="v-messages__wrapper"
          />
        </div>
      </div>
    </div>
  </div>
   
  <button
    class="v-btn v-btn--block v-btn--disabled v-btn--large theme--light primary--text"
    disabled="disabled"
    type="submit"
  >
    <div
      class="v-btn__content"
    >
      
    Connect
  
    </div>
  </button>
</form>
`;
